curated_transformers/layers/attention.py

```diff
@@ -407,15 +542,27 @@ def forward(
             )
 
         if _TORCH_SDP.get():
+            attn_mask = (
+                None if combined_mask is None else combined_mask.logit_mask(query.dtype)
+            )
+
+            # Add AliBi to the logit mask
+            if self.use_alibi and attn_mask is not None:
+                assert combined_mask is not None
+                assert self.attention.linear_biases is not None
+                bool_mask = combined_mask.bool_mask
+                biases = self.attention.linear_biases.calculate_biases(key.size(-2)).to(
```

Comment:
```suggestion
                biases = self.attention.linear_biases.calculate_biases(query.size(-2)).to(
```

Since the number of key heads can be 1.

Reply:
We're using the key length here, so the number of heads shouldn't matter.